packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,119 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def build_or_augment_dataframe(args, attrables, array_attrables):
+    """
+    Constructs a dataframe and modifies `args` in-place.
+
+    The argument values in `args` can be either strings corresponding to
+    existing columns of a dataframe, or data arrays (lists, numpy arrays,
+    pandas columns, series).
+
+    Parameters
+    ----------
+    args : OrderedDict
+        arguments passed to the px function and subsequently modified
+    attrables : list
+        list of keys into `args`, all of whose corresponding values are
+        converted into columns of a dataframe.
+    array_attrables : list
+        argument names corresponding to iterables, such as `hover_data`, ...
+    """
+
+    # We start from an empty DataFrame except for the case of functions which
+    # implicitely need all dimensions: Splom, Parcats, Parcoords
+    # This could be refined when dimensions is given
+    df = pd.DataFrame()
+
+    if "dimensions" in args and args["dimensions"] is None:
+        if args.get("data_frame") is None or args["data_frame"] is None:
+            raise ValueError(
+                "No data were provided. Please provide data either with the `data_frame` or with the `dimensions` argument."
+            )
+        else:
+            df_args = args["data_frame"]
+            df[df_args.columns] = df_args[df_args.columns]
+
+    # Valid column names
+    df_columns = (
+        args["data_frame"].columns if args.get("data_frame") is not None else None
+    )
+    group_attrs = ["symbol", "line_dash"]
+    for group_attr in group_attrs:
+        if group_attr in args:
```

Comment:
this case is already handled below, so you may as well just add `symbol` and `line_dash` in attrables always at call-time?

Reply:
ok done